### DISCUSSION TITLE
Предложения по доработке эмулятора в SmartApp Studio

### DISCUSSION THREAD

**Discussion by NikolayPakhomov** [sberdevices/salute-issues#181] [SmartMarket Studio]
Здесь можно оставить свои предложения по доработке эмулятора.

**Comment by snyuryev** [comment]
Конечно канвас аппы!

**Reply by awinogradov** [reply]
А зачем тебе запуск канвасов в эмуляторе, когда ты можешь локально запустить свой канвас и жить счастливо?

**Reply by snyuryev** [reply]
В целом да, можно и локально запустить. 
Если думать в сторону развития самого эмулятора, то можно было бы:
- показывать как выглядит канвас в приложении Салют (Сбер онлайн) со всеми отступами.
- показывать как выглядит канвас на девайсах Сбер бокс и Сбер портал - если запускать локально, то достаточно сложно понять где начинается область для саджестов (мб это больше вопрос к отображению саджестов локально).

**Comment by snyuryev** [comment]
Хотелось бы возможность масштабирования виртуального девайса (телефон хочется сделать чуть больше)

**Comment by Genenko-Svetlana** [comment]
@xoxulin @filomat @sasha-tlt

**Comment by dbf256** [comment]
Кажется, что в эмуляторе не поддерживаются карточки (cards), выглядят как кнопки минимального размера:

![image](https://user-images.githubusercontent.com/2550295/113470517-e97c3480-945e-11eb-82ff-4238311a9d2e.png)

**Reply by sasha-tlt** [reply]
Можете прислать json карточки? чтобы мы могли у себя воспроизвести?

**Reply by dbf256** [reply]
`{"card": {"items": [{"top_text": {"text": "1. Аконкагуа", "typeface": "button2", "type": "text_cell_view", "max_lines": 10, "text_color": "default"}, "bottom_text": {"text": "", "typeface": "button2", "text_color": "default", "max_lines": 10, "margins": {"top": "0x"}, "type": "text_cell_view"}, "type": "greeting_grid_item", "actions": [{"text": "1"}], "paddings": {"top": "6x", "right": "6x", "bottom": "0x", "left": "6x"}}, {"top_text": {"text": "2. Эверест", "text_color": "default", "typeface": "button2", "type": "text_cell_view", "max_lines": 10}, "bottom_text": {"text": "", "typeface": "button2", "text_color": "default", "max_lines": 10, "margins": {"top": "0x"}, "type": "text_cell_view"}, "type": "greeting_grid_item", "actions": [{"text": "2"}], "paddings": {"top": "6x", "right": "6x", "bottom": "0x", "left": "6x"}}, {"top_text": {"text": "3. Килиманджаро", "text_color": "default", "typeface": "button2", "type": "text_cell_view", "max_lines": 10}, "bottom_text": {"text": "", "typeface": "button2", "text_color": "default", "max_lines": 10, "margins": {"top": "0x"}, "type": "text_cell_view"}, "type": "greeting_grid_item", "actions": [{"text": "3"}], "paddings": {"top": "6x", "right": "6x", "bottom": "0x", "left": "6x"}}, {"top_text": {"text": "4. Мак-Кинли", "text_color": "default", "typeface": "button2", "type": "text_cell_view", "max_lines": 10}, "bottom_text": {"text": "", "typeface": "button2", "text_color": "default", "max_lines": 10, "margins": {"top": "0x"}, "type": "text_cell_view"}, "type": "greeting_grid_item", "actions": [{"text": "4"}], "paddings": {"top": "6x", "right": "6x", "bottom": "0x", "left": "6x"}}], "type": "grid_card", "item_width": "small", "columns": 2}}`

**Comment by Genenko-Svetlana** [comment]
@xoxulin @filomat @sasha-tlt

**Reply by snyuryev** [reply]
просьба добавить текстовый ввод запроса (для портала и сбербокса)

**Comment by Genenko-Svetlana** [comment]
@xoxulin @filomat @sasha-tlt

**Reply by snyuryev** [reply]
Есть еще такая идея - автоматически открывать последний респонс от бэкенда.
(сейчас если выбрать конкретный респонс в списке и приходит новый, то необходимо руками нажимать на последний респонс, чтобы открыть его)

Это будет более наглядно - видеть результат на экране и респонс.
